### DISCUSSION TITLE
Support TypeScript?

### DISCUSSION THREAD

**Discussion by drewsteinacher** [dotnet/interactive#3207] [Ideas]
Although I'm aware that plain JavaScript is supported, has anyone considered supporting TypeScript in via something like [ts-node](https://github.com/TypeStrong/ts-node)?

**Comment by jonsequitur** [comment]
I think it's been suggested, and if someone wanted to do the dev work, we'd be happy to advise, suggest design directions, review code, etc. It's not something we're currently planning, though. Wrapping ts-node in a .NET Interactive kernel would probably not be a huge amount of work for basic code execution functionality. I don't have a sense of how much additional work it would be to add support for code completions and variable sharing.

It's worth noting that _where code runs_ is as important for a kernel as what language the code is in. The current JavaScript kernel isn't a NodeJS kernel. It runs JavaScript in a browser or webview. A TypeScript kernel that did the same might be useful as well.

**Comment by marke-apexit** [comment]
There is a BSD-3 licensed TypeScript kernel for Jupyter on GitHub.  I know nothing about the dotnet / interactive and polyglot notebooks implementation, but that seems like a potential starting place.

see: https://github.com/winnekes/itypescript

**Comment by jonsequitur** [comment]
Existing Jupyter kernels can be integrated into .NET Interactive by following the instructions here: https://github.com/dotnet/interactive/blob/main/docs/adding-jupyter-kernels.md

**Comment by mika76** [comment]
I think there would be quite a bit of interest in this, plus Typescript is a Microsoft project isn't it - isn't there someone on that team who could help?
